polaris-core/src/main/java/org/apache/polaris/core/context/RealmScope.java

```diff
@@ -16,12 +16,14 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.polaris.service.config;
+package org.apache.polaris.core.context;
 
-import org.apache.polaris.core.PolarisConfigurationStore;
+import jakarta.inject.Scope;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
 
-/** Interface allows injection of a {@link PolarisConfigurationStore} */
-public interface ConfigurationStoreAware {
-
-  void setConfigurationStore(PolarisConfigurationStore configurationStore);
-}
+@Scope
+@Retention(java.lang.annotation.RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
```

Comment:
```suggestion
@Target({ElementType.TYPE, ElementType.METHOD, ElementType.FIELD})
```

Reply:
hmm - right now, the bean scope is defined at either the factory or the bean class itself. I don't know that the dependents ought to decide the scope of the bean its depending on